File: android/src/main/java/com/drumpicker/DrumPickerDefaults.kt
package com.drumpicker

import android.graphics.Color

/**
 * Visual defaults only. Row labels always come from the JS `items` prop.
 */
internal object DrumPickerDefaults {
  const val ITEM_HEIGHT_DP = 44f
  const val VISIBLE_ITEM_COUNT = 5
  const val TEXT_SIZE_SP = 20f
  const val SELECTED_TEXT_SIZE_SP = 22f
  const val SELECTION_INDICATOR_HEIGHT_DP = 1f

  val TEXT_COLOR: Int = Color.parseColor("#8E8E93")
  val SELECTED_TEXT_COLOR: Int = Color.parseColor("#1C1C1E")
  val SELECTION_INDICATOR_COLOR: Int = Color.parseColor("#D1D1D6")
}

File: src/drumPickerDefaults.ts
/**
 * Default prop values for `DrumPicker`, shared between the native
 * (`DrumPicker.native.tsx`) and web (`DrumPicker.tsx`) implementations so the
 * two platforms can never silently drift apart.
 *
 * These mirror the visual defaults baked into the native views
 * (`DrumPickerDefaults` in Kotlin / Swift). Keep them in sync if those change.
 */
export const DRUM_PICKER_DEFAULTS = {
  selectedIndex: 0,
  itemHeight: 44,
  visibleItemCount: 5,
  textColor: '#8E8E93',
  selectedTextColor: '#1C1C1E',
  textSize: 20,
  selectedTextSize: 22,
  showSelectionIndicator: true,
  selectionIndicatorColor: '#D1D1D6',
  selectionIndicatorHeight: 1,
  backgroundColor: 'transparent',
  itemBackgroundColor: 'transparent',
  containerBackgroundColor: 'transparent',
  hapticFeedback: false,
  accessibilityLabel: 'Picker',
} as const;
